### PR TITLE
PV2-491 merge payment due with required payments

### PR DESCRIPTION
**Pull request by kuzinator**

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs [hunk 1]
```diff
@@ -1,46 +1,36 @@
...
 using SFA.DAS.Payments.RequiredPayments.Domain;
 using SFA.DAS.Payments.RequiredPayments.Domain.Entities;
 using SFA.DAS.Payments.RequiredPayments.Model.Entities;
 
-namespace SFA.DAS.Payments.RequiredPayments.Application.UnitTests.Application.Handlers
+namespace SFA.DAS.Payments.RequiredPayments.Application.UnitTests.Application.Processors
 {
     [TestFixture]
-    public class IncentivePaymentDueEventHandlerTest
+    public class FunctionalSkillEarningsEventProcessorTest
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs]: We don't seem to have any test to ensure that no payment is made for earning in future period

File: src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/EarningEventHandler.cs [hunk 2]
```diff
@@ -32,7 +34,7 @@ public class PaymentDueEventHandler : IHandleMessages<PaymentDueEvent>
             this.lifetimeScope = lifetimeScope;
         }
 
-        public async Task Handle(PaymentDueEvent message, IMessageHandlerContext context)
+        public async Task Handle(EarningEvent message, IMessageHandlerContext context)
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/EarningEventHandler.cs]: Log Job Id is not updated when writing logs - (ESFA.DC.Logging.ExecutionContext)lifetimeScope.Resolve<IExecutionContext>();

Inject IExecutionContext into the constructor instead

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs [hunk 3]
```diff
@@ -85,60 +75,91 @@ public async Task TestHandleNullEvent()
...
-            var actualRequiredPayment = await handler.HandlePaymentDue(paymentDue, paymentHistoryCacheMock.Object, CancellationToken.None);
+            var actualRequiredPayment = await eventProcessor.HandleEarningEvent(earningEvent, paymentHistoryCacheMock.Object, CancellationToken.None);
 
             // assert
             Assert.IsNotNull(actualRequiredPayment);
-            Assert.AreEqual(1, actualRequiredPayment.AmountDue);
-            Assert.AreEqual(paymentDue.LearningAim.Reference, actualRequiredPayment.LearningAim.Reference);
+            Assert.AreEqual(1, actualRequiredPayment.Count);
+            Assert.AreEqual(1, actualRequiredPayment.First().AmountDue);
+            Assert.AreEqual(earningEvent.LearningAim.Reference, actualRequiredPayment.First().LearningAim.Reference);
```
Comment (samuel-dev-dba) [inline src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs]: We should also confirm that we have the right PriceEpisodeIdentifier

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs (hunk 1 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs]: added

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs (hunk 3 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Processors/FunctionalSkillEarningsEventProcessorTest.cs]: done

File: src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/EarningEventHandler.cs (hunk 2 above)
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsProxyService/Handlers/EarningEventHandler.cs]: this wasn't part of this task but done anyway

**Comment by ESFA-Pawan** [comment]
New pull request created for release branch - https://github.com/SkillsFundingAgency/das-payments-V2/pull/123

**Pull request merged**